### PR TITLE
Add Arbitrary instances

### LINKED ISSUE #54: Backfill Arbitrary instances to `string` modules
Depends on #49

### LINKED ISSUE #55: Backfill Arbitrary instances to `number` and `date` modules
Depends on #49

### PR DESCRIPTION
**Pull request by skeate**
Closes #54
Closes #55

Couple things unrelated to those in particular:

1. I moved the tests to match the structure of `src`
2. github is showing a few of the moved files as deleted/new files, but I did move them properly in Git so changes should track as expected

rest will be noted by relevant code

### CODE REVIEW

File: src/internal/util.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/** @internal */
+export const base64Encode = (s: string): string =>
+  Buffer ? Buffer.from(s).toString('base64') : btoa(s)
```
Comment (skeate) [inline src/internal/util.ts:3]: to work in both node and web

File: src/number/NegativeFloat.ts [hunk 2]
```diff
@@ -70,7 +73,7 @@ export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, NegativeFloat
  * @category Refinements
  */
 export const isNegativeFloat = (f: number): f is NegativeFloat =>
-  typeof f === 'number' && G.number.is(f) && f < 0 && f >= Number.MIN_SAFE_INTEGER
+  typeof f === 'number' && G.number.is(f) && f < 0 && f >= -Number.MAX_VALUE
```
Comment (skeate) [inline src/number/NegativeFloat.ts:76]: `MIN`/`MAX_SAFE_INTEGER` is nowhere close to the largest float that can be represented; copy/paste oversight perhaps?

File: src/string/JWT.ts [hunk 3]
```diff
@@ -113,3 +117,21 @@ export const Type: SchemableParams1<t.URI> = pipe(t.string, t.refine(isJWT, 'JWT
...
+  const header = base64Encode(JSON.stringify({ alg: 'none', typ: 'JWT' })).replaceAll(
+    /[=+/]/g,
+    c => (c === '/' ? '_' : c === '+' ? '-' : c === '=' ? '' : c)
+  )
+  const payload = base64Encode(JSON.stringify(json)).replaceAll(/[=+/]/g, c =>
+    c === '/' ? '_' : c === '+' ? '-' : c === '=' ? '' : c
+  )
+  const signature = ''
+  return [header, payload, signature].join('.')
+}) as Arb.Arbitrary<JWT>
```
Comment (skeate) [inline src/string/JWT.ts:137]: _really_ wanted these to be legit JWTs but I can't find a decent cross-platform crypto library that isn't async

File: tests/date/SafeDate.test.ts [hunk 4]
```diff
@@ -1,62 +1,70 @@
-import { Decoder, isSafeDate, Eq, Guard, TaskDecoder, Type } from '../src/date/SafeDate'
+import * as SafeDate from '../../src/date/SafeDate'
```
Comment (skeate) [inline tests/date/SafeDate.test.ts:1]: I mostly did this so I could use an easy snippet to add all the `Arbitrary` tests, but also I think it just makes more sense in these files

**Review by jacob-alford — APPROVED** [review]
💯  Really excited to get all these in!

**Review by jacob-alford — APPROVED** [review]
🥇

**Pull request merged**